statsmodels/tsa/arima_model.py

```diff
@@ -926,8 +916,10 @@ def fit(self, start_params=None, trend='c', method="css-mle",
         self.exog = exog    # overwrites original exog from __init__
 
         # (re)set names for this model
-        self.exog_names = _make_arma_names(self.data, k_trend, (k_ar, k_ma),
-                                           self.exog_names)
+        if self._first_call:
+            self.exog_names = _make_arma_names(self.data, k_trend,
+                                               (k_ar, k_ma), self.exog_names)
+            self._first_call = False
```

Comment:
Working on #5833 it turns out the failing tests there caused by incorrectly retained state, so affected by this design decision.

Since these depend on k_trend, won't these need to be invalidated on subsequent calls?

It would be a PITA to get the indirection to work correctly, but the proper place for this would be on the Results object.

Reply:
This is probably wrong solution, despite passing tests.  Given the overall design problems, simplest correct method is to fully recreate exog_names on each call to fit.

Reply:
Fully recreating does seem like the least-bad of a bunch of bad options.  Minimizes impact on future fixes, of which plenty will be needed until we address the underlying problems.